Describe new RBAC role setup (#3570)
kyma-project/kyma: docs/security/docs/03-04-roles-in-kyma.md

---
title: Roles in Kyma
type: Details
---

> Kyma uses roles and groups to manage access in the cluster. Every cluster comes with 5 predefined roles which give the assigned users different level of permissions suitable for different purposes.

Reviewer: ```suggestion
Kyma uses roles and groups to manage access in the cluster. Every cluster comes with five predefined roles which give the assigned users a different level of permissions.
```

Reviewer: -> https://docs.microsoft.com/en-us/style-guide/numbers#numerals-vs-words

Author: Replaced all numerals with a word

> These roles are defined as ClusterRoles and use the Kubernetes mechanism of aggregation, which allows to combine multiple ClusterRoles into a single ClusterRole. Using the aggregation mechanism allows to efficiently manage access to Kubernetes resources and Kyma-specific resources.

Reviewer: ```suggestion
These roles are defined as ClusterRoles and use the Kubernetes mechanism of aggregation which combines multiple ClusterRoles into a single ClusterRole. Use the aggregation mechanism to efficiently manage access to Kubernetes and Kyma-specific resources.
```

Reviewer: Hope the meaning is preserved? If not just pls change to `allows you to` ...

Author: I don't think the mechanism combines - the person who defines the roles combines them using the mechanism of aggregation. I'd stick with the original.

Reviewer: In that case `you` is missing here

>**NOTE:** Read [this](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#aggregated-clusterroles) Kubernetes documentation to learn more about the aggregation mechanism used to define Kyma roles.

You can assign any of the predefined roles to a user or to a group of users in the context of:  

>   - the entire cluster, creating a [ClusterRoleBinding](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#rolebinding-and-clusterrolebinding)

Reviewer: ```suggestion
  - The entire cluster by creating a [ClusterRoleBinding](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#rolebinding-and-clusterrolebinding)
```

>   - a specific Namespace, creating a [RoleBinding](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#rolebinding-and-clusterrolebinding)

Reviewer: ```suggestion
  - A specific Namespace by creating a [RoleBinding](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#rolebinding-and-clusterrolebinding)
```

The predefined roles, arranged in the order of increasing access level, are:

| Role | Description |
| --- | --- |

> | **kyma-essentials** | The basic role required to allow the users to see the Console UI of the cluster. This role doesn't give the user rights to manipulate any resources. |

Reviewer: ```suggestion
| **kyma-essentials** | The basic role required to allow the users to see the Console UI in the cluster. This role doesn't give the user rights to modify any resources. |
```

> | **kyma-view** | The role allowing to get and list Kubernetes resources and Kyma-specific resources. |

Reviewer: ```suggestion
| **kyma-view** | The role allowing to list Kubernetes and Kyma-specific resources. |
```

Author: "get" and "list" are kubernetes resources and that's why they're both listed here

Reviewer: The reason I removed `get`  was that it was used as a description here, not as a command. Thus, in the overall meaning, isn't it just about listing their details (read access permissions as the opposite to write access permissions)?

| **kyma-edit** | The role allowing to edit Kyma-specific resources.  |

> | **kyma-developer** | The role created for developers who build implementations using Kyma. Allows to edit, get, and list Kubernetes resources and Kyma-specific resources. |

Reviewer: ```suggestion
| **kyma-developer** | The role created for developers who build implementations using Kyma. It allows you to list and edit Kubernetes and Kyma-specific resources. |
```

Reviewer: again, `allowing to` everywhere ;)

> | **kyma-developer** | The role created with the developers who build implementations using Kyma. Allows to edit, get, and list Kubernetes resources and Kyma-specific resources. |

Reviewer 2: ```suggestion
| **kyma-developer** | The role created for developers who build implementations using Kyma. Allows to edit, get, and list Kubernetes resources and Kyma-specific resources. |
```
